nucleo_wb55rg: test/spi/spi_loopback build failure

**Describe the bug**
When building `test/spi/spi_loopback` with nucleo_wb55rg, build failure happens:
```
zephyr/drivers/dma/libdrivers__dma.a(dmamux_stm32.c.obj): in function `dmamux_stm32_init':
/local/mcu/zephyrproject/zephyr/drivers/dma/dmamux_stm32.c:155: undefined reference to `k_malloc'
```

**To Reproduce**
Steps to reproduce the behavior:
`sanitycheck -b -p nucleo_wb55rg -T tests/drivers/spi`

**Expected behavior**
Test should build

--- a/drivers/dma/Kconfig.stm32
+++ b/drivers/dma/Kconfig.stm32
@@ -39,11 +39,4 @@ config DMA_STM32_SHARED_IRQS
 	help
 	  Enable shared IRQ support on devices where channels share 1 IRQ.
 
-if DMAMUX_STM32
-
-config HEAP_MEM_POOL_SIZE
-	default 1024
-
-endif # DMAMUX_STM32
-
 endif # DMA_STM32

--- a/soc/arm/st_stm32/common/Kconfig.defconfig.series
+++ b/soc/arm/st_stm32/common/Kconfig.defconfig.series
@@ -75,6 +75,10 @@ if DMA
 config DMA_STM32
 	default y
 
+config HEAP_MEM_POOL_SIZE
+	default 1024
+	depends on DMAMUX_STM32
+
 endif # DMA
 
 endif # SOC_FAMILY_STM32

